torchrl/data/rlhf/dataset.py

```diff
@@ -0,0 +1,467 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+from __future__ import annotations
+
+import importlib.util
+import os
+from pathlib import Path
+
+from typing import Sequence, Type
+
+import torch
+
+from tensordict import TensorDict, TensorDictBase
+
+from tensordict.tensordict import NestedKey
+from torchrl.data import TensorDictReplayBuffer, TensorStorage
+from torchrl.data.replay_buffers import SamplerWithoutReplacement
+from transformers import AutoTokenizer
+
+_has_datasets = importlib.util.find_spec("datasets") is not None
+
+
+def create_or_load_dataset(
```

Comment:
what about calling this just "load_dataset" and maybe add an argument such as "exists" in mkdir?

Reply:
My suggestion/discussion is whether we could rename this to load_dataset without changing it's current behaviour and add an argument to specify the behaviour we want in case it does not exists.

Reply:
Im fine with renaming it, though I think that users should be informed that this function caches the results and the name says that quite explicitly.
I'm not sure I see what the extra arg should do. If it does not exist we must load it, and if it does we should use the cache. What extra control should we give?

Reply:
just a flag like exists_ok of mkdir that raises if it does not exists or something similar to `from_disk` but control whether we want to store cache or not.
I don't have strong opinion on this I am just reasoning if it's really necessary such a long name (even if definitely more explicit) or if it will be enough to say that we cache in the doc.